src/Workspaces/Core/Portable/Rename/Renamer.RenameSymbolDocumentAction.cs

```diff
@@ -0,0 +1,141 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using System.Linq;
+using System;
+using System.IO;
+
+namespace Microsoft.CodeAnalysis.Rename
+{
+    public static partial class Renamer
+    {
+        /// <summary>
+        /// Action that will rename a type to match the current document name. Works by finding a type matching the origanl name of the document (case insensitive) 
+        /// and updating that type. https://github.com/dotnet/roslyn/issues/43461 tracks adding more complicated heuristics to matching type and file name. 
+        /// </summary>
+        internal sealed class RenameSymbolDocumentAction : RenameDocumentAction
+        {
+            private readonly AnalysisResult _analysis;
+
+            private RenameSymbolDocumentAction(
+                AnalysisResult analysis)
+                : base(ImmutableArray<ErrorResource>.Empty)
+            {
+                _analysis = analysis;
+            }
+
+            public override string GetDescription(CultureInfo? culture)
+                => string.Format(WorkspacesResources.ResourceManager.GetString("Rename_0_to_1", culture ?? WorkspacesResources.Culture)!, _analysis.OriginalDocumentName, _analysis.NewDocumentName);
+
+            internal override async Task<Solution> GetModifiedSolutionAsync(Document document, OptionSet optionSet, CancellationToken cancellationToken)
```

Comment:
does this need to be internal?

Reply:
RenameDocumentAction is a public class, so I think this does need to be internal since it's an override but still needs to be called outside of this class.